### DISCUSSION TITLE
Clicking on an X-Y plane (e.g. plot, image) in a survey

### DISCUSSION THREAD

**Discussion by rgmerck** [surveydown-dev/surveydown#111] [Feature Requests]
### Description

Hi again, one of the thing that is probably possible thanks to Shiny would be to support questions where the answer is not really a regular user input as text or a radio button, but a position on a picture or a selection on a picture. There are many psychological tests where respondents are asked to put "where are you on this map" or "there are two X and Y axes for these dimensions, where would you put yourself in there?". Shiny has reactive elements that can handle the detection of a single click on a picture or even a selection on a picture, and would return coordinates or a range of coordinates for the data. 

Probably not a super common use case, but that's something most survey systems do not offer, and something I would definitely use myself.

**Comment by jhelvy** [comment]
Very cool, and yes that could definitely be done. But I'm not sure if this would be something to support in `sd_question()` as this would likely require a lot of custom settings based on the context (e.g. is the image a ggplot, a static image, etc.). 

What could be done is you could set this up where in your `server()` function in the `app.R` file you could have an image rendered and stored in the output, e.g.:

```r
output$map <- renderPlot({ something here }) # Or some other render function
```

Then in your `survey.qmd` file you put where you want that to be displayed just like in a normal shiny app, either using `sd_output()` or some other shiny output function. 

Then if there are some things that get captured in the server like the position they clicked, you could store those values to the data using `sd_store_value()`. 

Haven't tried it but that's a starting point if you want to give it a shot.

**Comment by pingfan-hu** [comment]
Hi @rgmerck , I leave a comment here so this discussion appears on my notification box. This is a great idea. I'll put it on my list and work on it in the near future.

**Comment by pingfan-hu** [comment]
Hi @rgmerck , here is the solution: we have a new function called `sd_question_custom()`, which is a placeholder for you to define a customized question.

[This demo](https://github.com/surveydown-dev/demos/tree/main/leaflet-map) is a showcase of the leaflet map question utilizing the `sd_question_custom()` function. You can take a look and play with it.

The **map question** looks like this:
<img width="853" alt="image" src="https://github.com/user-attachments/assets/90b57d36-ba8d-46cb-99cd-c6590da8fc50" />

[This demo](https://github.com/surveydown-dev/demos/tree/main/plotly) is a showcase of the plotly X-Y plane question utilizing the `sd_question_custom()` function. You can take a look as well.
The **plot** looks like this:
<img width="849" alt="image" src="https://github.com/user-attachments/assets/2d3bf845-b143-40d9-ac54-f69aa9d985b4" />

The demo codes have package installation codes prepared on the top of the `app.R` file so you can install the necessary packages by one click.
